client/lib/url/omit-url-params.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsy } from 'utility-types';
+
+/**
+ * Internal dependencies
+ */
+import { determineUrlType, URL_TYPE } from './url-type';
+import format from './format';
+
+const BASE_URL = 'http://__domain__.invalid';
+
+/**
+ * Removes given params from a url.
+ *
+ * @param  url URL to be cleaned
+ * @param  paramsToOmit The collection of params or single param to reject
+ * @return Url less the omitted params.
+ */
+export default function omitUrlParams( url: Falsy, paramsToOmit: string | string[] ): null;
+export default function omitUrlParams( url: URLString, paramsToOmit: string | string[] ): URLString;
```

Comment:
That all sounds very reasonable. I should point out that these types and overloads were already mostly in place. I believe I made some small tweaks to them, but the groundwork was already there, and since it seemed solid, I kept it in place.